File: domain/src/main/java/com/beat/domain/performance/repository/PerformanceRepository.java
package com.beat.domain.performance.repository;

import java.util.List;
import java.util.Optional;

import com.beat.domain.performance.domain.Genre;
import com.beat.domain.performance.domain.Performance;

public interface PerformanceRepository {
	Optional<Performance> findById(Long id);

	List<Performance> findAll();

	Performance save(Performance performance);

	List<Performance> findByGenre(Genre genre);

	List<Performance> findByUserId(Long userId);

	void deleteById(Long id);
}

File: core/domain/src/main/kotlin/com/beat/domain/performance/repository/PerformanceRepository.kt
package com.beat.domain.performance.repository

import com.beat.domain.performance.model.Performance
import java.util.*

@JvmSuppressWildcards
interface PerformanceRepository {
    fun findById(id: Long?): Optional<Performance>

    fun lockById(id: Long?): Optional<Performance>

    fun save(performance: Performance): Performance

    fun deleteById(id: Long?)
}
